assets/app/scripts/controllers/deployments.js

```diff
@@ -85,6 +120,222 @@ angular.module('openshiftConsole')
       }
     }
 
+    $scope.startLatestDeployment = function(deploymentConfigName) {
+      var deploymentConfig = $scope.deploymentConfigs[deploymentConfigName];
+
+      // increase latest version by one so starts new deployment based on latest
+      var req = {
+        kind: "DeploymentConfig",
+        apiVersion: "v1",
+        metadata: deploymentConfig.metadata,
+        spec: deploymentConfig.spec,
+        status: deploymentConfig.status
+      };
+      if (!req.status.latestVersion) {
+        req.status.latestVersion = 0;
+      }
+      req.status.latestVersion++;
+
+      // update the deployment config
+      DataService.update("deploymentconfigs", deploymentConfigName, req, $scope).then(
+        function() {
+            $scope.alerts = [
+            {
+              type: "success",
+              message: "Deployment #" + req.status.latestVersion + " of " + deploymentConfigName + " has started.",
+            }
+          ];
+        },
+        function(result) {
+          $scope.alerts = [
+            {
+              type: "error",
+              message: "An error occurred while starting the deployment.",
+              details: getErrorDetails(result)
+            }
+          ];
+        }
+      );
+    };
+
+    $scope.retryFailedDeployment = function(deploymentConfigName, deploymentName) {
+      var deployment = $scope.deploymentsByDeploymentConfig[deploymentConfigName][deploymentName];
+      var req = deployment;
+
+      // delete the deployer pod as well as the deployment hooks pods, if any
+      DataService.list("pods", $scope, function(list) {
```

Comment:
I definitely agree, but this is exactly what the command line tools are doing - should be fixed on the server and then we remove it from both the cli and web console, but having both doing different things will quickly lead to inconsistencies.